reframe/core/buildsystems.py

```diff
@@ -908,7 +916,7 @@ def _create_env_cmds(self):
 
         self.environment = 'rfm_spack_env'
         self._auto_env = True
-        return [f'spack env create -d {self.environment}']
+        return [f'spack env create -d {self.environment} {" ".join(self.env_create_opts)}']
```

Comment:
This adds an additional trailing whitespace if `env_create_opts` is empty and that's why unit tests are failing.